pandas/tools/merge.py

```diff
@@ -1264,13 +1264,21 @@ def flip(xs):
 
         # a "by" parameter requires special handling
         if self.left_by is not None:
```

Comment:
is this also changed for ``right_by``? any way to make this a touch more DRY. (don't spend a lot of time as this is pretty complicated, but any simplification would be great).